controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentOrder.java

```diff
@@ -32,61 +27,19 @@
  */
 public class DeploymentOrder {
 
-    private static final Logger log = Logger.getLogger(DeploymentOrder.class.getName());
-
-    private final Controller controller;
-    private final Clock clock;
+    private final Supplier<SystemName> system;
```

Comment:
Why not just `SystemName` and pass that in the constructor as this no longer needs the entire controller?

Reply:
Some tests rely on setting the system of the controller post-creation, which made this impossible.

Reply:
Ok, then I suggest passing `Supplier<SystemName>` to the constructor instead of the entire controller.